repository/SmalltalkCI-Core.package/SCIMetacelloLoadSpec.class/instance/basicLoadProjectOn..st

```diff
@@ -12,17 +12,21 @@ basicLoadProjectOn: aSmalltalkCI
 		baseline: self baseline;
 		configuration: self configuration;
 		version: self version.
-
+	self user ifNotEmpty: [:currentUser |
```

Comment:
Does it make sense to only provide `user` and not or `password` (or vice-versa)? I mean can you access a private repo by only providing a `password` or `user`? I'd be surprised if that works for `user`, but it may work for `password` if it's set to some API token. If some combination is invalid (e.g. when only a `user` is provided), is Metacello failing the smalltalkCI build with a good error message?

Reply:
> but it may work for password if it's set to some API token.

Yep, I'm using this a lot because it saves me the time to define another CI secret for my username. See https://github.com/Metacello/metacello/blob/master/repository/Metacello-Platform.squeak.package/MCFilesystemFetchOnlyRepository.class/class/sitePassword.st. :-)  
For username only, I'm not sure if there any scenarios, but I would rather not forbid this without a good reason (and if at all, this should be rather implemented in Metacello, shouldn't it?).

> is Metacello failing the smalltalkCI build with a good error message?

It depends. For an invalid password, for example, it says:

```
gofer repository error: 'GoferRepositoryError: Error downloading xxx to yyy.zip. Server said: {"message":"Not Found","documentation_url":"https://docs.github.com/rest/reference/repos#download-a-repository-archive"}'...ignoring
```

In this case, the repository was private, so the API does not distinguish between inaccessible and inexistent for privacy reasons, but otherwise, the message would be something like "Not authorized".

Reply:
Ok, it should really be Metacello's job to validate the input anyway :)